File: hw/ona/u2/hw13ToOurProgrammingLanguage/python/CarView.py
print("CarView module loaded")

import sys
sys.path.append('C:\\Segundo-Tercero\\02_Parcial\\python')

from db_connection import Connection  # Ensure this matches the actual file name
from datetime import datetime


class CarView:
    def __init__(self):
        self.conexion = Connection()
        self.mongo_client, self.db = self.conexion.crear_conexion()

    def save_car_details(self):
        if not self.mongo_client:
            print("Failed to connect to MongoDB.")
            return

        # Input car details
        id = int(input("Enter car ID: "))
        model = input("Enter car model: ")
        color = input("Enter car color: ")
        sale_date_str = input("Enter sale date (MM/DD/YYYY): ")

        try:
            sale_date = datetime.strptime(sale_date_str, "%m/%d/%Y")
        except ValueError:
            print("Invalid date format. Please use MM/DD/YYYY.")
            return

        # Create car document
        car_doc = {
            "id": id,
            "model": model,
            "color": color,
            "saleDate": sale_date
        }

        # Insert car document into MongoDB
        try:
            result = self.db.cars.insert_one(car_doc)
            print("Car details saved successfully to MongoDB.")
        except Exception as e:
            print(f"Error saving car details to MongoDB: {str(e)}")

# Example usage
if __name__ == "__main__":
    view = CarView()
    view.save_car_details()

File: hw/ona/u2/hw11MongoDBFile/MongoDBFiles/src/espe/edu/es/mongoDB/view/CarView.java
package espe.edu.es.mongoDB.view;

import espe.edu.es.mongoDB.modell.Car;
import espe.edu.es.mongoDB.util.Conexion;
import org.bson.Document;

import java.util.Scanner;
public class CarView {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        
        System.out.print("Enter car ID: ");
        int id = scanner.nextInt();
        scanner.nextLine(); // Consume newline character
        System.out.print("Enter car model: ");
        String model = scanner.nextLine();
        System.out.print("Enter car color: ");
        String color = scanner.nextLine();
        System.out.print("Enter sale date (MM/DD/YYYY): ");
        String saleDate = scanner.nextLine();

        
        Car car = new Car(id, model, color, saleDate);

        
        Conexion conexion = new Conexion();
        conexion.crearConexion(); // Conectar a la base de datos

        
        Document carDoc = new Document() ;
        carDoc.append("id", car.getId())
                .append("model", car.getModel())
                .append("color", car.getColor())
                .append("saleDate", car.getSaleDate());

        try {
            conexion.getDataB().getCollection("cars").insertOne(carDoc);
            System.out.println("Car details saved successfully to MongoDB.");
        } catch (Exception e) {
            System.out.println("Error saving car details to MongoDB: " + e.getMessage());
        } finally {
            conexion.getMongo().close(); // Cerrar la conexión a MongoDB
        }

        scanner.close();
    }
}
